### PR TITLE
RFR: Add book add/edit pages for admin user

### PR DESCRIPTION
**Pull request by josh-byster**
## Status: 

:construction: In development

## Description
A few sentences describing the overall goals of the pull request's commits.
This PR builds out the add book and edit pages along with the basic scaffolding for the admin pages (routing, basic styling, etc.)
Related issues: #<number>
Related PRs: #121

## Screenshots

image

image

### CODE REVIEW

File: frontend/src/components/AdminBookForm.js [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+          reader_url: this.props.currentBook.reader_url,
+          year: this.props.currentBook.year,
+          grade: this.props.currentBook.grade,
+          id: this.props.currentBook.id
+        },
+        () => {
+          this.testImage();
+          // Can't reuse the testBookURL method since it relies on an event
+          var patt = new RegExp(
+            /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/
```
Comment (Davidw1339) [inline frontend/src/components/AdminBookForm.js]: nice regexes!!

File: frontend/src/components/AdminBookForm.js [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            placeholder="Ex: http://book.com/file.pdf"
+          />
+          <FormFeedback invalid="true">
+            That doesn't look like a valid link.
+          </FormFeedback>
+          <FormFeedback valid>That link looks good!</FormFeedback>
+        </FormGroup>
+
+        <FormGroup>
+          <Button
```
Comment (Davidw1339) [inline frontend/src/components/AdminBookForm.js]: Nice reuse :recycle:  for conditional rendering of button based on editing or creating book

File: frontend/src/components/AdminBookForm.js [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+  testBookURL = e => {
+    this.handleChange(e);
+    var patt = new RegExp(
+      /[-a-zA-Z0-9@:%_\+.~#?&//=]{2,256}\.[a-z]{2,4}\b(\/[-a-zA-Z0-9@:%_\+.~#?&//=]*)?/
+    );
+    this.setState({ bookURLValid: patt.test(e.target.value) });
+  };
+
+  testImage = () => {
```
Comment (XuanLi-Leong) [inline frontend/src/components/AdminBookForm.js]: Super sweet!

File: frontend/src/pages/admin/AdminAddBookPage.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        <Row>
+          <Col lg="2">
+            <AdminNavigator />
+          </Col>
+          <Col lg="6" className="admin-home">
+            <h1>Add Book</h1>
+            <hr />
+            {this.state.success ? (
+              <Alert color="success">
+                Book was successfully created. Would you like to{' '}
```
Comment (XuanLi-Leong) [inline frontend/src/pages/admin/AdminAddBookPage.js]: Love this message

**Comment by XuanLi-Leong** [comment]
Remaining todos:
#### Admin
- [ ] Need to be an admin user
#### CSV Upload
- [ ] Add csv template
- [ ] Call POST function in `handleSubmit` #120 
- [ ] Accept multiple files instead of just one. Ideally would show processing state of each file. (i.e. whether call to `book_from_csv` was successful, for each file)

**Pull request merged**